src/Metadata/AttributeMetadataFactory.php

```
@@ -135,6 +140,50 @@ private function getPropertyMetadataList(ReflectionClass $reflectionClass): arra
         return array_values($properties);
     }
 
+    /** @return list<CallbackMetadata> */
+    private function getPostHydrateCallbacks(ReflectionClass $reflection): array
+    {
+        $methods = [];
+
+        foreach ($reflection->getMethods() as $reflectionMethod) {
+            if ($reflectionMethod->isStatic()) {
+                continue;
```

I think we should handle this differently, for dx reasons. If a user right now would use the attribute on a static method, it just wont work - but we could adjust it here to notify the user *why* it is not working. The cost for that is a performance optimization which i assume is tiny.

If we check the method for `static` after we know there is a attribute on that, we can throw a dedicated exception which will tell the user that this is not working. WDYT?

---

We have the same thing with the properties and other attributes. If we decide to throw an exception, then i would do it everywhere and in another PR.